deku-derive/src/macros/deku_read.rs

```diff
@@ -449,11 +467,35 @@ fn check_bitsize_container(
                     "f64" => core::mem::size_of::<f64>() * 8,
                     &_ => panic!("unsupported"),
                 };
-                if max_bit_size < bit_size {
-                    panic!(
-                        "field `{}` of type `{}` with bit size: {}, not big enough for bits = {}",
-                        field_ident, ident, max_bit_size, bit_size
-                    );
+                check_bitsize(max_bit_size, type_bit_size, field_ident, type_ident);
+            }
+        }
+    }
+}
+
+/// Convert `ctx::BitSize` into usize and check with `check_bitsize_bits_bytes`
+fn check_bitsize_ctx(
```

Comment:
This is clever (and probably still useful to have!), however because it's an expression, we can't assume it's contents. I think a dynamic check is still required.

```rust
struct DekuTest {
    #[deku(ctx = "deku::ctx::BitSize(std::mem::size_of::<u16>())")]
    field_a: u8
}
```

Reply:
Didn't think about that, how would I check that dynamically at compile time?

Reply:
That's the main issue, I'm not sure of how this can be done. It's almost like code needs to be generated, then compiled & executed as opposed to just executed in the proc-macro